### ISSUE TITLE
Feature request: Saved custom reports

### ISSUE THREAD

**Issue by vipin2079** [snipe/snipe-it#4616]
Currently we have select a filed to generate a report.If we can save report format with customize name then we can run again and again.

**Comment by wilsonlspacheco** [comment]
Very helpful feature !
Can save custom reports !
Our company export data to make a excel VBA apps to convert supplier's invoices to our database software. 
We vote this feature too! +1
We are a hosted costumer.

**Comment by BHollowell** [comment]
+1 on this for me as well. We export specific data for our Finance team on a monthly basis and would be helpful to save this report for quicker turnaround.

**Comment by markbrule** [comment]
+1

**Comment by BHollowell** [comment]
Looks like stalebot closed this automatically, sooooo....bump?

**Comment by smksar** [comment]
Oh, was about to post a request of the same... +1 indeed!

**Comment by esspeecy410** [comment]
+1 for me, being able to rerun reports using only certain custom fields without having to go through the checklist every time would be a massive help for us.

**Comment by tvreeland-securitymetrics** [comment]
Having the ability to save the format for a custom report would help out greatly. Our accounting department needs to have an inventory report every month and right now they have to go through and create the same format any time that they want to look at the report.

**Comment by LC-cchristiansen** [comment]
+1 from me as well.

Being able to re-run saved customized reports quickly would be hugely beneficial, not only from a time saving measure, but also for a consistent report aspect.

**Comment by bradrush** [comment]
Scheduled email of custom reports too please

**Comment by bradrush** [comment]
Yes this is still relevant, and should have been implemented by now. It is so tedious recreating the same reports repeatedly. Having an automated report emailed on a schedule would be hugely time saving.

Please implement this basic feature!

**Comment by LC-cchristiansen** [comment]
Please keep open

Get Outlook for Android<https://aka.ms/ghei36>






[http://cdn.laughlin.com/laughlin/media/public/img/lc-signatures/Griffin_67_72.png]<http://www.laughlin.com/>


Colin Christiansen
System Support Engineer
Laughlin Constable

414-270-7215  |  laughlin.com<http://www.laughlin.com/>




[http://cdn.laughlin.com/laughlin/media/public/img/lc-signatures/FB-Social_22.png]<https://www.facebook.com/LaughlinConstable/>    [http://cdn.laughlin.com/laughlin/media/public/img/lc-signatures/Twitter-Social_22.png] <https://twitter.com/laughlinagency>    [http://cdn.laughlin.com/laughlin/media/public/img/lc-signatures/Instagram-Social_22.png] <https://www.instagram.com/laughlinconstable>     [http://cdn.laughlin.com/laughlin/media/public/img/lc-signatures/LinkedIn-Social_22.png] <https://www.linkedin.com/company/laughlin-constable>     [http://cdn.laughlin.com/laughlin/media/public/img/lc-signatures/YouTube-Social_22.png] <https://www.youtube.com/user/LaughlinConstable>

CONFIDENTIALITY NOTICE: This email and any attachment(s) ("Message") is confidential and may contain privileged information intended only for the named recipient(s). If you are not the intended recipient(s), you are hereby notified that any review, retention, dissemination, distribution, and or copying of this Message is strictly prohibited. If you receive this Message in error, or are not the named recipient(s), please notify the sender at the email address above, delete this email from your computer, and destroy any copies in any form immediately. Any defamatory, discriminatory, harassing, threatening or obscene statements in this Message are contrary to Laughlin Constable policy and outside the sender's scope of employment. Laughlin Constable disclaims responsibility or liability as to such communication, and the sender shall be personally liable therefore.

On Sat, Jan 19, 2019 at 11:17 PM -0600, "stale[bot]" <notifications@github.com<mailto:notifications@github.com>> wrote:


Is this still relevant? We haven't heard from anyone in a bit. If so, please comment with any updates or additional detail.
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Don't take it personally, we just need to keep a handle on things. Thank you for your contributions!

-

**Comment by erenustun** [comment]
@wilsonlspacheco Hi Wilson! Since you are a hosted user. Did the snipe it customized your build to add this feature mentioned in this thread? I'd appreciate your reply since I'm looking to get this feature by the Snipe IT Team or try to code it myself. :) Thank you!

**Comment by wilsonlspacheco** [comment]
> @wilsonlspacheco Hi Wilson! Since you are a hosted user. Did the snipe it customized your build to add this feature mentioned in this thread? I'd appreciate your reply since I'm looking to get this feature by the Snipe IT Team or try to code it myself. :) Thank you!

We are no longer a hosted user, due to the your slow development of add-ons and new functionalities.

**Comment by erenustun** [comment]
> > @wilsonlspacheco Hi Wilson! Since you are a hosted user. Did the snipe it customized your build to add this feature mentioned in this thread? I'd appreciate your reply since I'm looking to get this feature by the Snipe IT Team or try to code it myself. :) Thank you!
> 
> We are no longer a hosted user, due to the your slow development of add-ons and new functionalities.

Thanks a lot for the quick reply!

**Comment by LC-cchristiansen** [comment]
Still relevant

**Comment by tvreeland-securitymetrics** [comment]
Yes still relevant

**Comment by bradrush** [comment]
Yes, this is still relevant and can you please stop closing it having not even looked at it in many many months.


Brad Rush
Systems & Implementation Engineer

E  brad.rush@seequent.com
M  +64 275 48 5519
T  +64 3 961 1031

Enabling better decisions about earth, environment, and energy challenges

[https://blog.leapfrog3d.com/wp-content/uploads/2017/11/line-break-resize.png]

[https://blog.leapfrog3d.com/wp-content/uploads/2017/11/Seequent-Logo-Dispatch_Master-Lockup-Charcoal-resized.png]<http://www.seequent.com/>

**Comment by bradrush** [comment]
Yes, still relevant. We haven't heard of any progress either, how bout that?

Thanks,
Brad Rush

Brad Rush
Systems & Implementation Engineer

E  brad.rush@seequent.com
M  +64 275 48 5519
T  +64 3 961 1031

Enabling better decisions about earth, environment, and energy challenges

[https://blog.leapfrog3d.com/wp-content/uploads/2017/11/line-break-resize.png]

[https://blog.leapfrog3d.com/wp-content/uploads/2017/11/Seequent-Logo-Dispatch_Master-Lockup-Charcoal-resized.png]<http://www.seequent.com/>


________________________________

**Comment by twisted3motions** [comment]
This would be great.

**Comment by yayadrian** [comment]
+1 for this feature

**Comment by tvreeland-securitymetrics** [comment]
This is still relevant just waiting for it to actually be addressed.

**Comment by lito777** [comment]
This feature would be a dealbreaker! Looking forward for it

**Comment by JSheldon42** [comment]
+1 on this feature.

**Comment by yayadrian** [comment]
Any news on the progress of this or if it is even on the roadmap?

**Comment by vedder26** [comment]
+1 on this feature.

**Comment by namarie23** [comment]
+20000 this feature his highly desired.

**Comment by JSheldon42** [comment]
Still relevant, how can we light a fire under this one?

**Comment by snipe** [comment]
This is not going to be worked on until v5 is released.

**Comment by yayadrian** [comment]
Agree v5 is important once that’s done then would love this feature
request.

**Comment by lito777** [comment]
When is v5 supposed to be done?

**Comment by snipe** [comment]
> When is v5 supposed to be done?

When it's ready, and not before then.

We're shooting for next week or the week after, but folks are welcome to test on the develop branch to speed up that process.

**Comment by coryputnamch** [comment]
+1 on this feature.  Emailing reports to Finance each week is one fo the top 5 requirements for an inventory system for us.

**Comment by namarie23** [comment]
Still relevant

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by yayadrian** [comment]
Yes please.

**Comment by geeje** [comment]
Still relevant

**Comment by hesske** [comment]
The ability to save custom-built reports and schedule them is definitely a need

**Comment by arranreid** [comment]
Need this

**Comment by smalekzadeh** [comment]
This should be added to the Snipe as there are lots of interests in saving custom reports.

**Comment by jpguyon52** [comment]
We are also looking for this feature. We are using it tru snipe-it.io

**Comment by ghost** [comment]
Yes! And the ability to schedule & email those reports.

**Comment by GamingSanctum** [comment]
Bump. I wrote a powershell script that takes my chromebook inventory report from SnipeIT and applies changes to specified fields in the Google Admin Console via GAM. I would appreciate if I could save my custom report and I would LOVE if I could automate it to email or, even better, to a network location so I do not have to manually run a report before running my script.

**Comment by rdegennaro** [comment]
I've been writing views directly against the DB for the same reason of updating GAC.  Especially helpful during MCAS season and for DRC/ACCESS/ELL testing (both tests used in Mass.).  I also have "sanity checks" for common things missed on updating the ever-changing "current state" of hardware.

I'd love it if my techs didn't have to go into the DB and could just click on the report.  Now they have to log into phpMyAdmin with DB level users I've created.  Not the most secure, I know.  But needs must when the devil drives.  :-)

**Comment by jonnyiompo** [comment]
Yes we would like this feature

**Comment by rbuzzell** [comment]
I would like to add a 👍 to this feature as well.

**Comment by pearlofwisdom** [comment]
Created Github account just to add my +1 here!  Even just saving the selected fields and filter criteria would be great!  

ETA:  I'm frankly surprised that this request has been lingering for several years.  Seems clear to me that it's one many would like to see added.

**Comment by snipe** [comment]
@pearlofwisdom it's on the list, that's why it's marked "ready for dev".

I'd be happy to take a pull request for it, but nobody has offered one.

**Comment by pearlofwisdom** [comment]
@snipe Ah, I understand now, thanks.  Sorry, wish I could code!

**Comment by alpacalife** [comment]
Great feature idea!

**Comment by glanepulitzer** [comment]
Would love this, and possibility to select the order the columns are sorted into and schedule an emailed export!

Thanks!

**Comment by Mike-NP** [comment]
Would be great to see this added - is there an ETA?

**Comment by liamf91** [comment]
Certainly a +1

**Comment by SMistryASH** [comment]
+1 on the feature request. Has this not yet been implmented into any version release of the system? Would also like to be able to create custom reports by filtering on/by custom fields.

**Comment by danthi123** [comment]
+1 would really appreciate this feature

**Comment by DEKUJETA** [comment]
+1 would appreciate this feature. Has it been added yet?

**Comment by dtosun61** [comment]
+1 would really appreciate this feature

**Comment by Nimeshin** [comment]
+1 would really appreciate this feature

**Comment by BussolDev** [comment]
+1  c'mon!  You know you want to do it!

**Comment by cwolfram** [comment]
+1 from me as well!

**Comment by ChrisKtec** [comment]
And my ~~axe~~ +1

**Comment by gbadgett** [comment]
This would be a game changer for sure!

**Comment by Eamll** [comment]
+1 :)

**Comment by RobertD1985** [comment]
+1 but this is from 2017 and stil no option to save custom reports

**Comment by HesDead103** [comment]
+1 How is this so old and still not seen any progress?

**Comment by marcusmoore** [comment]
We're planning on introducing this in the near(ish) future. No definite timeline but it is in progress.

**Comment by HesDead103** [comment]
> We're planning on introducing this in the near(ish) future. No definite timeline but it is in progress.

Excited to hear this!

**Comment by Artapel** [comment]
bump

**Comment by marcusmoore** [comment]
This is still in the works but the dev team has been focusing on the v7 release. I expect this feature to land after the v7 release.

**Comment by websafe** [comment]
👍

**Comment by liamf91** [comment]
bump for this.

**Comment by lb1f** [comment]
Would love to see this too. Even if it was a case of having to store them manually in a directory on the server to make them available. I have to create asset reports for our servers, things like IP addresses etc I want included in the list, but I don't want to reduce the asset list view each time or create the report each time choosing each field manually.

**Comment by hermant1** [comment]
Would be great to have this. that we can send weekly/monthly reports to management of what was purchased/ given to users.

**Comment by gertz22** [comment]
+1 for me too

**Comment by HesDead103** [comment]
Any update on this feature? ❤️

**Comment by marcusmoore** [comment]
The first pass at this feature is awaiting internal review but it's a lot of code to check so it's taking a bit.

"The first pass" means templates that can be saved and easily _manually_ re-run. It doesn't include scheduled reports but that is something we would like to address in the future.

You can see the proposed changes in this PR #15714.

**Comment by marcusmoore** [comment]
The first pass mentioned above has been completed in #15714.
